File: sdks/sdkwork-mail-sdk/sdkwork-mail-sdk-python/sdkwork_mail_sdk/capability_catalog.py
from dataclasses import dataclass
from typing import Optional


@dataclass(frozen=True)
class MailCapabilityCatalogEntry:
    capabilityKey: str
    category: str
    surface: str


class MailCapabilityCatalog:
    entries = [
        MailCapabilityCatalogEntry("transport.connect", "required-baseline", "control-plane"),
        MailCapabilityCatalogEntry("transport.authenticate", "required-baseline", "control-plane"),
        MailCapabilityCatalogEntry("health", "required-baseline", "control-plane"),
        MailCapabilityCatalogEntry("smtp.send", "optional-advanced", "runtime-bridge"),
        MailCapabilityCatalogEntry("imap.sync", "optional-advanced", "runtime-bridge"),
        MailCapabilityCatalogEntry("imap.folder-sync", "optional-advanced", "runtime-bridge"),
        MailCapabilityCatalogEntry("imap.message-sync", "optional-advanced", "runtime-bridge"),
        MailCapabilityCatalogEntry("transport.retry", "optional-advanced", "control-plane"),
        MailCapabilityCatalogEntry("transport.pool", "optional-advanced", "control-plane"),
    ]


def get_mail_capability_catalog() -> list[MailCapabilityCatalogEntry]:
    return MailCapabilityCatalog.entries


def get_mail_capability_descriptor(capability_key: str) -> Optional[MailCapabilityCatalogEntry]:
    for entry in MailCapabilityCatalog.entries:
        if entry.capabilityKey == capability_key:
            return entry

    return None

File: sdks/sdkwork-mail-sdk/sdkwork-mail-sdk-go/Mailstandard/capability_catalog.go
package Mailstandard

type MailCapabilityCatalogEntry struct {
    CapabilityKey string
    Category      string
    Surface       string
}

type MailCapabilityCatalog struct{}

var mail_CAPABILITY_CATALOG = []MailCapabilityCatalogEntry{
    {CapabilityKey: "transport.connect", Category: "required-baseline", Surface: "control-plane"},
    {CapabilityKey: "transport.authenticate", Category: "required-baseline", Surface: "control-plane"},
    {CapabilityKey: "health", Category: "required-baseline", Surface: "control-plane"},
    {CapabilityKey: "smtp.send", Category: "optional-advanced", Surface: "runtime-bridge"},
    {CapabilityKey: "imap.sync", Category: "optional-advanced", Surface: "runtime-bridge"},
    {CapabilityKey: "imap.folder-sync", Category: "optional-advanced", Surface: "runtime-bridge"},
    {CapabilityKey: "imap.message-sync", Category: "optional-advanced", Surface: "runtime-bridge"},
    {CapabilityKey: "transport.retry", Category: "optional-advanced", Surface: "control-plane"},
    {CapabilityKey: "transport.pool", Category: "optional-advanced", Surface: "control-plane"},
}

func GetMailCapabilityCatalog() []MailCapabilityCatalogEntry {
    return append([]MailCapabilityCatalogEntry(nil), mail_CAPABILITY_CATALOG...)
}

func GetMailCapabilityDescriptor(capabilityKey string) *MailCapabilityCatalogEntry {
    for index := range mail_CAPABILITY_CATALOG {
        if mail_CAPABILITY_CATALOG[index].CapabilityKey == capabilityKey {
            return &mail_CAPABILITY_CATALOG[index]
        }
    }

    return nil
}
